// rastgele/index.js
const snekfetch = require('snekfetch')
;

class rastgele {

    /**
     * SFW CAT
     */
    async cat() {
        try {
            var {
                body
            } = await snekfetch
                .get('https://aws.random.cat/meow');
            return body.file;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW DOG
     */
    async dog() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://random.dog/woof.json');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW KISS
     */
    async kiss() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/kiss');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW HUG
     */
    async hug() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/hug');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW POKE
     */
    async poke() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/poke');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW COFFEE
     */
    async coffee() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://coffee.alexflipnote.xyz/random.json');
            return body.file;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW CUDDLE
     */
    async cuddle() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/cuddle');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW NEKO
     */
    async neko() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/neko');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW FOX GIRL
     */
    async foxgirl() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/fox_girl');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW TICKLE
     */
    async tickle() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/tickle');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW SLAP
     */
    async slap() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/slap');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW SMUG
     */
    async smug() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/smug');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW BAKA
     */
    async baka() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/baka');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW PAT
     */
    async pat() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/pat');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW LIZARD
     */
    async lizard() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/lizard');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW FEED
     */
    async feed() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/feed');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW KEMONOMIMI
     */
    async kemonomimi() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/kemonomimi');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * NSFW YURI
     */
    async yuri() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekos.life/api/v2/img/yuri');
            return body.url;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * NSFW THIGHS
     */
    async thighs() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekobot.xyz/api/v2/image/thighs');
            return body.message;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }

    /**
     * SFW KANNA
     */
    async kanna() {
        try {
            const {
                body
            } = await snekfetch
                .get('https://nekobot.xyz/api/v2/image/kanna');
            return body.message;
        } catch (err) {
            throw 'Error: ' + err.message;
        }
    }
}

module.exports = new rastgele();
